docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -0,0 +1,366 @@
+
+![](./images/locked.png)
+
+Any data structures published using snowplow-cli will automatically get this banner and have UI based editing disabled. It is a good idea to settle on one source of truth for each data structure to avoid potential conflicts.
+
+### Publish to production
+
+With our data structure deployed to develop and working as we expect we can safely publish it to production.
+
+```bash
+$ snowplow-cli ds publish prod
+```
+```
+3:00PM INFO publishing to prod from paths=[data-structures]
+3:00PM INFO will update file=data-structures/com.example/login.yaml local=1-0-0 remote=""
+3:00PM INFO all done!
+```
+:::note
+Data structures must be published to `dev` before they can be published to `prod`
+:::
+
+We have now seen how to create, validate and then publish a new data structure from the command line. Next we'll look at how to configure github workflows to run validation and publishing automatically for us.
+
+
+## Automating with github workflows
+
+### Set up repository
+
+We'll not go into the details of creating github repositories and initial commits here. The [github docs](https://docs.github.com/) do an excellent job of that already. The next few steps will assume a working github repository containing the directory and data structure we created in the previous section. It will have two branches named `main` and `develop` which should be in sync.
+
+### Publish to develop workflow
+
+We would like pushes to our `develop` branch to be automatically published to our [development](/docs/managing-data-quality/testing-and-qa-workflows/) environment. Github workflows can be [triggered](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow) by all kinds of repository events. The one we are interested in here:
+```yml
+on:
+  push:
+    branches: [develop]
+```
+
+With our trigger point worked out we need to complete a series of steps:
+1. configure snowplow-cli via environment variables provided as [github action secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions)
+2. checkout our repo
+3. install snowplow-cli
+4. run the `snowplow-cli ds publish dev` command as we did earlier
+
+The full action:
+
+```yml title=".github/workflows/publish-develop.yml"
+on:
+  push:
+    branches: [develop]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      SNOWPLOW_CONSOLE_ORG_ID: ${{ secrets.SNOWPLOW_CONSOLE_ORG_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_ID: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_SECRET: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_SECRET }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: curl -L -o snowplow-cli https://github.com/snowplow-product/snowplow-cli/releases/latest/download/snowplow-cli_linux_x86_64 && chmod u+x snowplow-cli
+
+      - run: ./snowplow-cli ds publish dev --managed-from $GITHUB_REPOSITORY
```

Comment:
Should we move the `--managed-from` to be an env variable as well?

Reply:
maybe? I don't think I feel strongly. It kind of does fit with things like org id, yeah, id be ok with it either way

Reply:
I think it makes sense, it makes the command a bit cleaner